Applications/Spire/Include/Spire/Spire/RangeInputModel.hpp

```diff
@@ -0,0 +1,36 @@
+#ifndef SPIRE_RANGE_INPUT_MODEL_HPP
+#define SPIRE_RANGE_INPUT_MODEL_HPP
+#include "Spire/Charting/ChartValue.hpp"
+
+namespace Spire {
+
+  //! Represents a model for a range input widget.
+  class RangeInputModel {
+    public:
+
+      //! Represents a histogram for a range input widget.
+      struct Histogram {
+      
+        //! A histogram where each element is a bin, and the value
+        //! of the element is the bin's frequency.
+        std::vector<int> m_histogram;
+
+        //! The highest frequency count amoung all the histogram's bins.
+        int m_highest_frequency;
+      };
+
+      //! Returns the smallest value among the data.
```

Comment:
Needs a virtual destructor.

Reply:
Even if the class doesn't have dynamically allocated members?

Reply:
Good question, but yes, even without any dynamically allocated members. Take some time to read about why it's important.

Reply:
From Effective C++: "The C++ language standard is unusually clear on this topic. When you try to delete a derived class object through a base class pointer and the base class has a non-virtual destructor (as EnemyTarget does), the results are undefined."

I found a lot of explanations, but _if correct_, this short and sweet explanation clarifies it.